p2panda-net/src/engine/engine.rs

```diff
@@ -326,20 +344,37 @@ where
         Ok(())
     }
 
+    /// Return a receiver for system events.
+    fn events(&mut self) -> broadcast::Receiver<SystemEvent<T>> {
```

Comment:
We're using a broadcast channel to allow for multiple receivers to be created (if needed). This means that all receivers will receive a sent event (not just one of them).